enclave/attest_plugin.c

```diff
@@ -69,8 +69,9 @@ static oe_result_t _wrap_with_header(
 
 oe_result_t oe_get_evidence(
     const oe_uuid_t* format_id,
-    const oe_claim_t* custom_claims,
-    size_t custom_claims_length,
+    bool wrap_with_header,
```

Comment:
> @dthaler @yentsanglee From an API design perspective, should this be a bit flag array instead to accommodate future extensibility? Essentially, it's one of the few extensibility points for the OE SDK itself in the attestation plugin infrastructure.

I agree a 4-byte flag is more extensible.

Reply:
I'm ok either way. In the current design, there OE SDK itself does not do much. It basically just chooses a plugin for the format ID, and invokes plugin entry points.

I don't even see much value in the attestation header wrapped by the OE SDK. Since it is optional, the verifier can't inspect the evidence to tell its format or if a header is present. it must learn from the attester via some other means. The header used to be valuable when it was mandatory.

So I'm also OK if `oe_get_evidence()` is changed to never wrap an attestation header. With this change, there would be no need of the `wrap_with_header` parameter, and `oe_verify_evidence()` would always need a valid `format_id`.

@dthaler and @yentsanglee what are your recommendations?

Reply:
> I don't even see much value in the attestation header wrapped by the OE SDK. Since it is optional, the verifier can't inspect the evidence to tell its format or if a header is present. it must learn from the attester via some other means. The header used to be valuable when it was mandatory.

The value proposition of OE SDK offering a format-ID embedded wrapper hasn't changed between the iterations of this PR; even in the existing behavior of mandatory headers, the implicit contract between the attester and verifier was that both had to be using the OE SDK. The design choice to expose the wrapping option simply makes that contract more explicit for the parties that wish to continue to use that behavior, because OE SDK is expanding its interop with non-OE SDK attesters and verifiers.

Reply:
@CodeMonkeyLeet Thanks! Agreed that the value proposition stays as you described.